certbot-dovecot/certbot_dovecot/dovecotparser.py

```diff
@@ -0,0 +1,102 @@
+"""Very low-level dovecot config parser based on pyparsing."""
+from pyparsing import (
+    alphanums, restOfLine,
+    Literal, Forward, Group, LineEnd, Optional,
+    OneOrMore, ParserElement, White, Word, ZeroOrMore
+)
+
+class DovecotParser(object):
+    # pylint: disable=expression-not-assigned
+    """A class that parses dovecot configuration with pyparsing."""
+
+    # pyparsing automatically uses whitespaces as delimeters for tokens.
+    # Here, we're setting default whitespace characters to ` ` and `\t`,
+    # excluding `\n` since a new line is used as a delimeter for dovecot
+    # configuration itself
+    ParserElement.setDefaultWhitespaceChars('')
+
+    newline = LineEnd()
+    same_line_space = Optional(Word(' \t').leaveWhitespace())
+    space = Optional(White().leaveWhitespace())
+
+    # Defining comments
+    comment = space + Literal('#') + restOfLine
+
+    # Defining key and value pair
+    key = Word(alphanums) + same_line_space + (Literal('='))
```

Comment:
Be careful with using `alphanums`-- it doesn't look like [Dovecot's parser](https://github.com/dovecot/core/blob/master/src/config/config-parser.c) restricts tokens to use alphanumeric characters.

Reply:
From what I see here, https://github.com/dovecot/core/blob/master/src/config/config-parser.c#L572, it basically accepts all characters except whitespace. Thanks for that note!